### PR TITLE
Architecture. Tokens. Contract Function Calling.

### PR DESCRIPTION
**Pull request by zweigraf**
- [WA-11]: Maintain ERC20 token addresses
- [WA-10]: Sign and broadcast transactions
- [WA-15]: Work on app architecture

Token balances for the current account are also shown per token, but currently no whitelisted tokens exist & also no filtering for 0 balances.

Secrets are maintained via Code Generation with Sourcery in a build phase. Locally, people can add a `env-vars.sh` file which will be used during compilation, on CI the variables are set directly. An `env-vars.sh.example` file is added which can be changed to the real keys.

Tokens are currently stored on the file system via Disk. Keys are stored in the Keychain using KeychainAccess. Tests are written with Quick & Nimble.

### CODE REVIEW

File: Heimdall/Ethereum/EtherRPC+TokenInfo.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    func tokenInfo(for address: String) -> Signal<Token, Error> {
+        let nameSignal = call(HumanStandardToken.Name.self, ofContractAt: address)
+            .map { $0.value }
+        let symbolSignal = call(HumanStandardToken.Symbol.self, ofContractAt: address)
+            .map { $0.value }
+        // We can use forcedInt here, as the return value is only Solidity.UInt8
+        let decimalsSignal = call(HumanStandardToken.Decimals.self, ofContractAt: address)
+            .map { $0.value.forcedInt }
+
+        return combineLatest(nameSignal, symbolSignal, decimalsSignal)
```
Comment (rmeissner) [inline Heimdall/Ethereum/EtherRPC+TokenInfo.swift:23]: this is doing 3 request, right? So if 1 request fails the combineLatest will also fail. You could do them in a bulk request

File: Heimdall/Ethereum/NonceProvider.swift [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+//  Heimdall
+//
+//  Created by Luis Reisewitz on 30.10.17.
+//  Copyright © 2017 Gnosis. All rights reserved.
+//
+
+class NonceProvider {
+    func nextNonce() -> UInt {
+        // FIXME: dummy nonce, need to increment and get from blockchain maybe?
+        return 12
```
Comment (rmeissner) [inline Heimdall/Ethereum/NonceProvider.swift:12]: Hmm should that be comited like that?

File: Heimdall/Features/AccountOverview/AccountOverviewViewController.swift [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    let accountLabelText: Property<String>
+    let balanceLabelText: Property<String>
+
+    let title = Property("Your Account")
+
+    let disposeBag = DisposeBag()
+
+    init(credentials: Credentials, rpc: EtherRPC) {
+        accountLabelText = Property("""
+            This is your account.
```
Comment (rmeissner) [inline Heimdall/Features/AccountOverview/AccountOverviewViewController.swift:24]: should that be localized?

File: Heimdall/Model/Token.swift [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    static var whitelisted: [Token] {
+        return [
+
+        ]
+    }
+}
+// MARK: - Equatable
+extension Token: Equatable {
+    static func == (lhs: Token, rhs: Token) -> Bool {
+        return lhs.address == rhs.address && lhs.symbol == rhs.symbol
```
Comment (rmeissner) [inline Heimdall/Model/Token.swift]: I think equals on the address should be enough

**Review by rmeissner — APPROVED** [review]

File: Heimdall/Ethereum/EtherRPC+TokenInfo.swift (hunk 1 above)
Comment (zweigraf) [reply Heimdall/Ethereum/EtherRPC+TokenInfo.swift:23]: As far as I can see, my RPC provider does not allow for bulk requests at the moment. Looking to change that.

File: Heimdall/Features/AccountOverview/AccountOverviewViewController.swift (hunk 3 above)
Comment (zweigraf) [reply Heimdall/Features/AccountOverview/AccountOverviewViewController.swift:24]: Yes. Like every other string. Will do that asap (but not in this PR).

File: Heimdall/Model/Token.swift (hunk 4 above)
Comment (zweigraf) [reply Heimdall/Model/Token.swift]: thanks

File: Heimdall/Ethereum/NonceProvider.swift (hunk 2 above)
Comment (zweigraf) [reply Heimdall/Ethereum/NonceProvider.swift:12]: Haven't really thought about this completely yet, so I would say yes.

**Pull request merged**